kingpin/actors/aws/cloudformation.py

```diff
@@ -803,9 +821,11 @@ class Stack(CloudFormationBaseActor):
         'region': (str, REQUIRED, 'AWS region (or zone) name, like us-west-2'),
         'role_arn': (str, None,
                      'The Amazon IAM Role to use when executing the stack'),
-        'template': (str, REQUIRED,
-                     'Path to the AWS CloudFormation File. http(s)://, '
-                     'file:///, absolute or relative file paths.'),
+        'template': (str, None,
+                     'Path to the AWS CloudFormation File. (ie file:///), '
+                     'absolute or relative file paths.'),
+        'template_bucket': (str, None, 'S3 bucket containing the CFN template'),
+        'template_key': (str, None, 'Key in S3 bucket containing the CFN template'),
```

Comment:
So after reviewing #507 as well, I think that we should split the difference. I prefer the model now of one parameter for the stack location (`template`) - and let that parameter take two value forms:

```yaml
template: my_stack.yaml
```

or
```yaml
template: s3://bucketname/key
```

The __init__ method should then handle determining whether or not its an inline template or its an S3-hosted template.. and from there it can go and validate the template appropriately. The rest of the code in this PR i like more (rather than asking the user to construct S3 urls). We can add a `tempate_region` optional parameter if the user wants to be explicit about where the bucket is, otherwise we do the get_bucket_location lookup for them.